x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexFieldNameTranslator.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.support;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public class SecurityIndexFieldNameTranslator {
```

Comment:
I added this class to support prefix-field name matching, since that's used in the query apikey api. Now I only use this for the query user api, where we only have exact field name mappings and they all happen to match exactly what's in the security index, so translations are not required. For future-proofing it could be good to keep this, but I also understand that it looks overly complex. 

To justify this I'm going to use this for the api key api too in a separate PR.

Let me know your thoughts.

Reply:
@javanna I don't think @jfreden had immediate plans to work on this, so a PR is more than welcome! @jfreden is on PTO for the next two weeks